randovania/resolver/bootstrap.py

```diff
@@ -222,6 +226,45 @@ def assign_pool_results(self, rng: Random, patches: GamePatches, pool_results: P
             pool_results.starting
         )
 
+    def all_artifact_locations(
+        self,
+        game: GameDescription,
+        config: BaseConfiguration,
+        game_specific_check: Callable[[PickupNode, BaseConfiguration], bool],
+    ) -> list[PickupNode]:
+        locations = []
+
+        for node in game.region_list.all_nodes:
+            if isinstance(node, PickupNode) and game_specific_check(node, config):
+                locations.append(node)
+
+        return locations
+
+    def pre_place_artifacts(
```

Comment:
since these are supposed to be generic, can we rename the functions to not reference artifacts?
`all_preplaced_item_locations` and `pre_place_items` perhaps?

Reply:
I would prefer for us to have a proper name for these end-game locking key items. Don't care which name, but some.

Reply:
MacGuffin?

Reply:
If I understood dunc correctly he also wanted to potentially use these for other items, hence being very generic with them.

Reply:
I would suggest something like `dumb_distribute_items()`. `all_artifact_locations()` is more specifically about artifacts. if we want to make it more generic the only thing I can think of is `all_macguffin_locations()` like Ty suggested, but I think that's better suited for another pr

Reply:
If the intention is really to use it also for non dna/keys/artifacts whatever, then we also shouldn't use macguffin.